api/v1alpha1/ext_proc_types.go

```diff
@@ -80,4 +80,28 @@ type ExtProc struct {
 	//
 	// +optional
 	ProcessingMode *ExtProcProcessingMode `json:"processingMode,omitempty"`
+
+	// Metadata defines options related to the sending and receiving of dynamic metadata.
+	// These options define which metadata namespaces would be sent to the processor and which dynamic metadata
+	// namespaces the processor would be permitted to emit metadata to.
+	// Users can specify custom namespaces or well-known envoy metadata namespace (such as envoy.filters.http.ext_authz)
+	// documented here: https://www.envoyproxy.io/docs/envoy/latest/configuration/advanced/well_known_dynamic_metadata#well-known-dynamic-metadata
+	// Default: no metadata context is sent or received from the external processor
+	//
+	// +optional
+	Metadata *ExtProcMetadata `json:"metadata,omitempty"`
+}
+
+// ExtProcMetadata defines options related to the sending and receiving of dynamic metadata to and from the
+// external processor service
+type ExtProcMetadata struct {
+	// AccessibleNamespaces are metadata namespaces that are sent to the external processor as context
+	//
+	// +optional
+	AccessibleNamespaces []string `json:"accessibleNamespaces,omitempty"`
+
+	// WritableNamespaces are metadata namespaces that the external processor can write to
+	//
+	// +optional
```

Comment:
can we use CEL or validation to avoid writing into filters EG is using (especially ext authz, jwt, and rbac) https://www.envoyproxy.io/docs/envoy/latest/configuration/advanced/well_known_dynamic_metadata#well-known-dynamic-metadata ?